### PR TITLE
DEVPROD-33792 Add Rate Limit settings to API config

### PR DESCRIPTION
**Pull request by shreeyachand**
DEVPROD-33792

### Description

A new admin-settings section that holds the actual rate-limit numbers so an operator can tune them at runtime (backend only). 

There are ten different fields - eight are (REST, GraphQL) x (human, service-account) x (burst, per-hour), defining a separate limit for each combination of query type, account type, and limit type. There's additionally a `GraphQLComplexityLimit` field that limits the "complexity" of a GraphQL query, and an `ElevatedUsersIDs` list that determines which users get 2x their tier's baseline limit. Zero/unset values for these numeric fields **disable** the corresponding limit.

*Currently the complexity limit is just a per query ceiling, but additional settings like a per-hour complexity quota may be added depending on analysis of query complexity. 

These settings are added to both the service layer and the REST/GraphQL layer. 

### Testing

Added tests in `config_api_test.go` covering two areas: 1) validation unit tests for RateLimitConfig (non-negative rate and complexity limits, burst not exceeding the hourly rate) and 2) a BSON round-trip test (mirroring tests for existing config structs) confirming that all fields persist and retrieve correctly through APIConfig.Set/Get. Note that user ID existence is not validated because unknown IDs are harmless no-ops since the TTL cache check will end up being a plain string membership test.

Additionally tested the admin settings REST routes and GraphQL queries locally, ensuring these ten settings and their values can be modified and returned by the appropriate endpoints.

### CODE REVIEW

**Comment by bynn** [comment]
could you test this by trying to get these settings using the rest route?

**Comment by shreeyachand** [comment]
> could you test this by trying to get these settings using the rest route?

ah i had actually not added these settings to the rest layer. confirmed that they are returned by the rest route now:
Screenshot 2026-06-08 at 4 11 55 PM

**Review by bynn — CHANGES_REQUESTED** [review]
I think in theory, this change would be functional. However, it doesn't follow our conventions for adding settings. 

Currently, these settings are saved in the API config which is the admin configurations for our own APIs. It doesn't make too much sense for the code to live where it is and would be better suited if it the rate limit config struct lived [here](https://github.com/evergreen-ci/evergreen/blob/31807b519bb5c194f5259e82a2fb461152e7180d/config.go#L65) like the other configs do.

**Comment by bynn** [comment]
Also instead of saying "human" could we change the wording to "user" and "service user"

File: graphql/schema/types/adminSettings/web.graphql [hunk 1]
```diff
@@ -1,3 +1,29 @@
...
+  restUserBurst: Int
+  restServicePerHour: Int
+  restServiceBurst: Int
+  graphqlUserPerHour: Int
+  graphqlUserBurst: Int
+  graphqlServicePerHour: Int
+  graphqlServiceBurst: Int
+  graphqlComplexityLimit: Int
+  elevatedUserIds: [String!]
+}
```
Comment (sophstad) [inline graphql/schema/types/adminSettings/web.graphql:25]: [nit] Could you alphabetize these fields?

**Review by sophstad — CHANGES_REQUESTED** [review]

File: gqlgen.yml [hunk 2]
```diff
@@ -43,6 +43,10 @@ models:
     model: github.com/evergreen-ci/evergreen/rest/model.APIapiConfig
   APIConfigInput:
     model: github.com/evergreen-ci/evergreen/rest/model.APIapiConfig
+  RateLimitConfig:
```
Comment (bynn) [inline gqlgen.yml]: I  think this should be put in the correct alphabetical order too

**Review by bynn — APPROVED** [review]
I have 1 change but everything else looks fine

**Review by sophstad — APPROVED** [review]

File: gqlgen.yml (hunk 2 above)
Comment (shreeyachand) [reply gqlgen.yml]: reordering
